clients/azuredevopsrepo/work_items.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2024 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuredevopsrepo
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/workitemtracking"
+
+	"github.com/ossf/scorecard/v5/clients"
+)
+
+var (
+	errSystemCreatedByFieldNotMap        = fmt.Errorf("error: System.CreatedBy field is not a map")
+	errSystemCreatedByFieldNotUniqueName = fmt.Errorf("error: System.CreatedBy field does not contain a UniqueName")
+	errSystemCreatedDateFieldNotString   = fmt.Errorf("error: System.CreatedDate field is not a string")
+)
+
+type (
+	fnQueryWorkItems func(
+		ctx context.Context,
+		args workitemtracking.QueryByWiqlArgs,
+	) (*workitemtracking.WorkItemQueryResult, error)
+
+	fnGetWorkItems func(
+		ctx context.Context,
+		args workitemtracking.GetWorkItemsArgs,
+	) (*[]workitemtracking.WorkItem, error)
+
+	fnGetWorkItemComments func(
+		ctx context.Context,
+		args workitemtracking.GetCommentsArgs,
+	) (*workitemtracking.CommentList, error)
+)
+
+type workItemsHandler struct {
+	ctx                 context.Context
+	repourl             *Repo
+	once                *sync.Once
+	errSetup            error
+	workItemsClient     workitemtracking.Client
+	queryWorkItems      fnQueryWorkItems
+	getWorkItems        fnGetWorkItems
+	getWorkItemComments fnGetWorkItemComments
+	issues              []clients.Issue
+}
+
+func (w *workItemsHandler) init(ctx context.Context, repourl *Repo) {
+	w.ctx = ctx
+	w.errSetup = nil
+	w.once = new(sync.Once)
+	w.repourl = repourl
+	w.queryWorkItems = w.workItemsClient.QueryByWiql
+	w.getWorkItems = w.workItemsClient.GetWorkItems
+	w.getWorkItemComments = w.workItemsClient.GetComments
+}
```

Comment:
let's explicitly clear `issues` too (we've had accumulation problems before)

```go
w.issues = nil
```

Reply:
Done!